WordPress/gutenberg
# Border Controls: Add placeholders and base styles

Depends on:
* #34178
* #33743
* #34061
* #34467

Alternative approach to #34131

## Description

As we convert the border supports to use the new Tools Panel (#33743) and configure default controls for each block (#34061), we want to prevent needing to enable border style as a default. This was previously done because the style needs to be selected in order for changes to color or width to make visible changes in the editor.

This PR makes use of the addition of an API to access global styles in the block editor (#34178) to solve this problem, and also add placeholders/defaults for each of the border attributes.

Changes:
* When a width or color is selected, the `solid` style will be automatically applied if none is selected/available from global styles
* Values from global styles are applied as placeholders for each border attribute

At the moment this PR also includes a fix for a bug in the border color control, to correctly check selected presets and display the selected color in the color indicator. Since this could be merged separately I also have a PR here with just those changes broken out: https://github.com/WordPress/gutenberg/pull/34467

## Notes
* This has a lot of dependencies. The changes can be viewed most easily compared against the #34178 branch. It can be rebased on #33743 to get the ToolsPanel. It may be easier to review after at least one of those is merged.

## How has this been tested?




I've tested this with just #34178, and also rebased on top of #33743 which converts the panel to the new ToolsPanel. It should work in both cases. I used the Group block to test.

* Test behavior of the placeholders for each of `width / style / color / radius` in the editor
    * Set values in your `theme.json`, something like this:
    ```
            "core/group": {
				"border": {
					"radius": {
						"topLeft": "20px",
						"topRight": "30px",
						"bottomLeft": "15px",
						"bottomRight": "45px"
					},
					"width": "2px",
					"color": "var(--wp--preset--color--dark-gray)",
					"style": "dashed"
				}
			}
		}
    ```
    * Set values in Global styles from the Site Editor --> By Block Type --> Group panel
    * Verify that the merged styles appear as placeholders in the Border panel for a Group block inserted in the editor
            
* Test the panel in Global styles to make sure it continues to function
    * Styles set in the theme should appear as placeholders in the Global styles panel

* Test setting a width or color when the style is not set
    * If style is not set, and there is no default value from global styles, a solid style should be automatically applied

## Screenshots 

Placeholders:
![border-panel-defaults](https://user-images.githubusercontent.com/63313398/131898730-d05c4785-ea93-4b99-856f-afe2ed8c86fa.gif)

Automatically setting border style when color/width selected:
![auto-set-border-style](https://user-images.githubusercontent.com/63313398/131898772-575f924a-804f-4a1b-a974-1a0f8a7d8303.gif)


## Types of changes





## Checklist:
